arithmetization/src/main/java/net/consensys/linea/zktracer/container/stacked/set/StackedSet.java

```diff
@@ -121,7 +122,8 @@ public <T> T[] toArray(@NotNull T[] a) {
 
   @Override
   public boolean add(E e) {
-    final boolean isNew = this.sets.peekLast().add(e);
+    final boolean isNew = Optional.ofNullable(this.sets.peekLast()).orElse(new HashSet<>()).add(e);
```

Comment:
We don't want this: there is no correct use case of the `StackedSet` where we never called `enter()`.

Either revert it or throw an exception, but don't sneakily insert a new set in it.

Reply:
Ok, I will revert it.